### PR TITLE
ACS-8541: Add userId in Questions and Answer API requests

### PR DESCRIPTION
**Pull request by krdabrowski**
https://hyland.atlassian.net/browse/ACS-8541

### CODE REVIEW

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        NodeRef currentUserNodeRef = personService.getPersonOrNull(authenticationService.getCurrentUserName());
+        String userId = ofNullable(currentUserNodeRef).map(NodeRef::getId).orElse(null);
+        return hxInsightClient.getAnswer(questionId, userId);
+    }
+
+    @Override
+    public String retryQuestion(String agentId, String questionId, String comments, Question question)
+    {
+        return hxInsightClient.retryQuestion(agentId, questionId, comments, question);
```
Comment (krdabrowski) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java:64]: I wonder if I should also add `userId` also here...? WDYT @Pawel-608? Probably yes 🤔

Comment (krdabrowski) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java:66]: Ok - it has to be added also here - fixed!

File: hxinsight-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-hxinsight-extension/context/service-context.xml [hunk 2]
```diff
@@ -158,7 +158,43 @@
...
     </bean>
 
-    <bean class="org.alfresco.hxi_connector.hxi_extension.rest.api.QuestionsEntityResource" />
+    <bean id="questionService" class="org.alfresco.hxi_connector.hxi_extension.service.QuestionServiceImpl">
+        <constructor-arg name="hxInsightClient" ref="hxInsightClient"/>
+        <constructor-arg name="authenticationService" ref="AuthenticationService"/>
+        <constructor-arg name="personService" ref="PersonService"/>
+    </bean>
+
+    <bean id="QuestionService" class="org.springframework.aop.framework.ProxyFactoryBean">
```
Comment (Pawel-608) [inline hxinsight-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-hxinsight-extension/context/service-context.xml:167]: I'm not sure if we need all these wrappers🤔

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+{
+    private final HxInsightClient hxInsightClient;
+    private final AuthenticationService authenticationService;
+    private final PersonService personService;
+
+    @Override
+    public String askQuestion(String agentId, Question question)
+    {
+        NodeRef currentUserNodeRef = personService.getPersonOrNull(authenticationService.getCurrentUserName());
+        question.setUserId(ofNullable(currentUserNodeRef).map(NodeRef::getId).orElse(null));
```
Comment (Pawel-608) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java:49]: maybe we could extract it to some method?

**Review by Pawel-608 — APPROVED** [review]
LGTM

File: hxinsight-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-hxinsight-extension/context/service-context.xml (hunk 2 above)
Comment (krdabrowski) [reply hxinsight-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-hxinsight-extension/context/service-context.xml]: I agree it's not necessary - `AuthenticationService` and `PersonService` has already configured security, transactions, etc. - wrappings removed

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java (hunk 3 above)
Comment (krdabrowski) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/QuestionServiceImpl.java]: Good tip - fixed!

**Pull request merged**